webpack/assets/javascripts/react_app/components/statistics/ChartBox.js

```diff
@@ -29,30 +30,29 @@ class ChartBox extends React.Component {
   }
 
   render() {
-    const { chart } = this.props;
-
-    const modalConfig = getLargePieChartConfig({
-      data: this.props.chart.data,
-      id: `${chart.id}Modal`,
-    });
-
+    const { chart, type } = this.props;
+    const components = {
+      donut: ForemanDonutChart,
+    };
+    const Chart = components[type];
     const tooltip = {
       onClick: this.onClick,
       title: this.props.tip,
       'data-toggle': 'tooltip',
       'data-placement': 'top',
+      style: { cursor: 'pointer' },
     };
     const onclickChartClicked =
-      chart.search && chart.search.match(/=$/) ? null : navigateToSearch.bind(null, chart.search);
+      chart.search && chart.search.match(/=$/)
+        ? null
+        : navigateToSearch.bind(null, chart.search);
 
-    const _chart = (
-      <PieChart
+    const _chart = (config = 'regullar') => <Chart
+        config={config}
         key={`${this.props.chart.id}-chart`}
-        data={this.props.chart.data}
+        data={chart.data ? Immutable.asMutable(chart.data) : {}}
```

Comment:
what was the thought process behind this ? how would a children/c3 change the data?

Reply:
looks like the issue occurs in `names` object
`c3` is trying to change it trough [here](https://github.com/c3js/c3/blob/69f975cfcf54626a36566a8b8236f63aec001951/c3.js#L5983)